### ISSUE TITLE
[ASK] Get recommendation

### ISSUE THREAD

**Issue by wahyudierwin** [lpeska/CABPR#1]
Hi @lpeska , hope that you are doing well.

I came across this great repository when I'm looking for hybrid version of BPR. But, I can not find a function to serve / get recommendation given a single user id or a user preference. My questions are:

1. How to store the trained model?
1. How to get recommendation? Preferably without filtering already seen items.

Thanks!

**Comment by lpeska** [comment]
Hi wahyudierwin,

sorry, the code is quite messy and it was coded for tensorflow 0.x, so I cannot guarantee it will work today. Plus, I only implemented the evaluation in a way to be reported in the paper (https://www.researchgate.net/publication/325916813_Hybrid_recommendations_by_content-aligned_Bayesian_personalized_ranking), so it is not very suitable for the tasks you have...

Nonetheless, the easiest way to store the trained model would be to store the CABPR object as a pickle (i.e. the "model" variable from the main loop). During training, several matrices and vectors are trained depending on the method variant, so alternatively you can store just the self.user_factors, self.item_factors, self.item_bias, self.sim_lambda.

As for how to get the recommendation for a single user, I would require to modify the evaluation() method a bit: it basically iterates through all test set data supplied in the argument, put estimated ratings into the scores array and then evaluates the scores w.r.t. the ground truth. So the only thing you need to do is to change what is tested, return the scores and remove the actual evaluation. 
To be more specific, alter "test_data" variable so that it contains pairs of (current_user, item) for all items you want to get prediction. Then remove rows 858-870 and put return scores there

To be completely honest, given the obsolete version of tensorflow and my messy coding in general, I would recommend to re-implement the version of the algorithm you're interested.  You can check the paper for details - I would recommend the CABPR-L2-NN variant. But it is up to you of course.

Best regards,
Ladislav
